### PR TITLE
Improve employee movements

### PR DESCRIPTION
**Pull request by mertwole**

### CODE REVIEW

File: JamGame/Assets/Scripts/Employee/EmployeeImpl.cs [hunk 1]
```diff
@@ -46,42 +44,33 @@ private enum State
...
-
         private NeedProvider.PlaceInWaitingLine placeInWaitingLine = null;
 
         [MinMaxSlider(0, 10, true)]
         [SerializeField]
         private Vector2 keepDistanceFromNextInLine = new(1.0f, 1.5f);
 
+        private void OnEnable()
+        {
+            controller = controller != null ? controller : GetComponent<ControllerImpl>();
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Employee/EmployeeImpl.cs:62]: Can you just `GetComponent` In Awake?

File: JamGame/Assets/Scripts/Employee/EmployeeModifiers.cs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            }
+
+            if (to_remove == -1)
+            {
+                Debug.LogError("Failed to unregister NeedModifierEffect: Not registered");
+                return;
+            }
+
+            employee.UnregisterModifier(registeredModifiers[to_remove].Item2);
+            GameObject.Destroy(registeredModifiers[to_remove].Item2.gameObject);
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Employee/EmployeeModifiers.cs:245]: Can you give descriptive names for `Item1` and `Item2`

File: JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs [hunk 3]
```diff
@@ -127,23 +134,23 @@ private void CorrectMovement()
...
-        public void Teleport(NeedProvider needProvider)
+        public void Teleport(Vector3 destination)
         {
-            transform.position = needProvider.transform.position + (agent.baseOffset * Vector3.up);
+            transform.position = destination + (agent.baseOffset * Vector3.up);
             state = State.BuildingPath;
         }
 
-        public float? ComputePathLength(NeedProvider need_provider)
+        public float? ApproximatePathLength(Vector3 destination)
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs:143]: Can you return Optional for descriptive return? I don't understand what is null is mean.

**Review by Kiuh — APPROVED** [review]

File: JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs (hunk 3 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs:143]: Discussed

**Pull request merged**